snuba/datasets/plans/single_storage.py

```diff
@@ -71,18 +81,25 @@ class SelectedStorageQueryPlanBuilder(StorageQueryPlanBuilder):
     def __init__(
         self,
         selector: QueryStorageSelector,
-        post_processors: Optional[Sequence[QueryProcessor]] = None,
+        post_processors: Optional[Sequence[PhysicalQueryProcessor]] = None,
     ) -> None:
         self.__selector = selector
         self.__post_processors = post_processors or []
 
     def build_plan(self, request: Request) -> StorageQueryPlan:
-        storage = self.__selector.select_storage(request.query, request.settings)
-        request.query.set_data_source(
+        storage, translator = self.__selector.select_storage(
```

Comment:
I hope to make this work for joins as well, allowing us to pick a storage per entity that we are joining and a translator for each so that each translator acts independently on the joined query translating the part related to the storage the translator is coupled with. Will do some experiments around this.